### PR TITLE
Merge existing changes for CT_v1.0.1 - Release 22.2 - Candidate

### PR DESCRIPTION
**Pull request by paulhowes-HMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-2027
https://tools.hmcts.net/jira/browse/EUI-3155
https://tools.hmcts.net/jira/browse/EUI-3550

### Change description ###
- Button aria-labels for accessibility (Previously PR 683)
- Accessibility amends for non-descriptive headings (Previously PR 656)
- Replace h2 tags with spans on case edit pages for accessibility (Previously PR 738)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by paul-graham — DISMISSED** [review]

**Comment by paulhowes-HMCTS** [comment]
Note - this was previously approved in https://github.com/hmcts/ccd-case-ui-toolkit/pull/738 however I have decided to create a new PR as we have changed from using Master to Release as the base and attempting rebasing was causing issues.

**Review by phillip-whitaker — DISMISSED** [review]

**Review by paul-graham — DISMISSED** [review]

**Review by paul-graham — DISMISSED** [review]

File: src/shared/components/search-filters/search-filters.component.html [hunk 1]
```diff
@@ -25,6 +26,6 @@
     </div>
   </ng-container>
 
-  <button [disabled]="!isSearchableAndSearchInputsReady()" type="button" class="button" (click)="apply()" (keyup.enter)="apply()" >Apply</button> &nbsp;&nbsp;&nbsp;
-  <button id="reset" type="button" (click)="reset()" class="button button-secondary">Reset</button>
+  <button [disabled]="!isSearchableAndSearchInputsReady()" type="button" class="button" (click)="apply()" (keyup.enter)="apply()" title="Apply filter" aria-label="Apply filter" >Apply</button> &nbsp;&nbsp;&nbsp;
```
Comment (paul-graham) [inline src/shared/components/search-filters/search-filters.component.html:29]: _Being annoying:_
What's that extra space at the end of the tag for? Future-proofing? Breathing room? Is the `aria-label` claustrophobic? 🤔

File: src/shared/components/workbasket-filters/workbasket-filters.component.html [hunk 2]
```diff
@@ -39,6 +39,6 @@ <h2 class="heading-h2" aria-label="Filters">Filters</h2>
     </div>
   </ng-container>
 
-  <button type="button" class="button" (click)="apply(true)" [disabled]="isApplyButtonDisabled()" >Apply</button> &nbsp;&nbsp;&nbsp;
-  <button type="button" (click)="reset()" class="button button-secondary">Reset</button>
+  <button type="button" class="button" (click)="apply(true)" [disabled]="isApplyButtonDisabled()" title="Apply filter" aria-label="Apply filter">Apply</button> &nbsp;&nbsp;&nbsp;
```
Comment (paul-graham) [inline src/shared/components/workbasket-filters/workbasket-filters.component.html:42]: _Still being annoying:_
So not a claustrophobic `aria-label` then. Unless it got some treatment or you're maliciously playing on its fears.

**Review by paul-graham — APPROVED** [review]

**Review by paul-graham — APPROVED** [review]

File: src/shared/components/search-filters/search-filters.component.html (hunk 1 above)
Comment (paulhowes-HMCTS) [reply src/shared/components/search-filters/search-filters.component.html]: Removed. No idea where that came from (my bad)

**Review by phillip-whitaker — APPROVED** [review]

**Comment by paulhowes-HMCTS** [comment]
Superceded by https://github.com/hmcts/ccd-case-ui-toolkit/pull/794

**Pull request closed without merging**